packages/backend/src/telemetry/service.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  Counter,
+  DiagConsoleLogger,
+  DiagLogLevel,
+  MetricOptions,
+  diag,
+  metrics
+} from '@opentelemetry/api'
+import { OTLPMetricExporter } from '@opentelemetry/exporter-metrics-otlp-grpc'
+import { Resource } from '@opentelemetry/resources'
+import {
+  MeterProvider,
+  PeriodicExportingMetricReader
+} from '@opentelemetry/sdk-metrics'
+
+import { ConvertError, RatesService } from '../rates/service'
+import { ConvertOptions } from '../rates/util'
+import { BaseService } from '../shared/baseService'
+
+export interface TelemetryService {
+  getOrCreateMetric(name: string, options?: MetricOptions): Counter
+  getServiceName(): string | undefined
+  getBaseAssetCode(): string
+  getBaseScale(): number
+  convertAmount(
+    convertOptions: Omit<ConvertOptions, 'exchangeRate' | 'destinationAsset'>
+  ): Promise<bigint | ConvertError>
+}
+
+interface TelemetryServiceDependencies extends BaseService {
+  serviceName: string
+  collectorUrls: string[]
+  exportIntervalMillis?: number
+  aseRatesService: RatesService
+  fallbackRatesService: RatesService
+  baseAssetCode: string
+  baseScale: number
+}
+
+export function createTelemetryService(
+  deps: TelemetryServiceDependencies
+): TelemetryService {
+  return new TelemetryServiceImpl(deps)
+}
+
+class TelemetryServiceImpl implements TelemetryService {
+  private serviceName: string
+  private meterProvider?: MeterProvider
+  private fallbackRatesService: RatesService
+  private aseRatesService: RatesService
+
+  private counters = new Map()
+  constructor(private deps: TelemetryServiceDependencies) {
+    diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.DEBUG)
+    this.serviceName = deps.serviceName
+    this.fallbackRatesService = deps.fallbackRatesService
+    this.aseRatesService = deps.aseRatesService
+
+    if (
+      deps.collectorUrls &&
+      Array.isArray(deps.collectorUrls) &&
+      deps.collectorUrls.length === 0
+    ) {
+      deps.logger.info(
+        'No collector URLs specified, metrics will not be exported'
+      )
+      return
+    }
+
+    this.meterProvider = new MeterProvider({
+      resource: new Resource({ 'service.name': 'RAFIKI_NETWORK' })
+    })
+
+    deps.collectorUrls.forEach((url) => {
+      const metricExporter = new OTLPMetricExporter({
+        url: url
+      })
+
+      const metricReader = new PeriodicExportingMetricReader({
+        exporter: metricExporter,
+        exportIntervalMillis: deps.exportIntervalMillis ?? 15000
+      })
+
+      this.meterProvider?.addMetricReader(metricReader)
+    })
+
+    metrics.setGlobalMeterProvider(this.meterProvider)
+  }
+
+  private createCounter(
+    name: string,
+    options: MetricOptions | undefined
+  ): Counter {
+    const counter = metrics.getMeter('Rafiki').createCounter(name, options)
```

Comment:
same quest as here basically: https://github.com/interledger/rafiki/pull/2299/files#r1474772665

Is it OK to hardcode `'Rafiki'` like this? I was not able to work out where exactly this is coming from.

Reply:
`metrics` is imported directly from the '`@opentelemetry/api`' package.
By calling `getMeter`, a new Meter named "Rafiki" in this case will be created and returned from the global meter provider.
It's just the name of the "Meter" that will be responsible of creating all our counter metrics.
The name is a required parameter, and I don't think it should be externaly configurable or something.

Moved it out of inline string definition into a const in the same file.